src/main/java/vazkii/patchouli/api/data/EntryBuilder.java

```diff
@@ -0,0 +1,194 @@
+package vazkii.patchouli.api.data;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+
+import vazkii.patchouli.api.PatchouliAPI;
+import vazkii.patchouli.api.data.page.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EntryBuilder {
+
+	private final CategoryBuilder parent;
+	private final ResourceLocation id;
+	private final String name;
+	private final String category;
+	private final String icon;
+	private final List<AbstractPageBuilder<?>> pages = new ArrayList<>();
+	private String advancement;
+	private String flag;
+	private Boolean priority;
+	private Boolean secret;
+	private Boolean readByDefault;
+	private Integer sortnum;
+	private String turnin;
+	private Map<ItemStack, Integer> extraRecipeMappings;
+
+	EntryBuilder(String id, String name, String icon, CategoryBuilder parent) {
+		this.id = new ResourceLocation(parent.getId().getNamespace(), id);
+		this.name = name;
+		this.category = parent.getId().getPath();
+		this.icon = icon;
+		this.parent = parent;
+	}
+
+	EntryBuilder(String id, String name, ItemStack icon, CategoryBuilder parent) {
+		this(id, name, PatchouliAPI.instance.serializeItemStack(icon), parent);
+	}
```

Comment:
You missed making these two constructors protected.

Reply:
thx totally missed that

Reply:
fixed